src/utilities.ts

```diff
@@ -151,3 +151,9 @@ export function getReqColor(groupName: string, onboardingData: AppOnboardingData
   }
   return reqGroupColorList[0];
 }
+
+// Determines whether the given element in a FireStoreSemester list is a Placeholder or not
+export const isPlaceholderCourse = (
+  element: FirestoreSemesterPlaceholder | FirestoreSemesterCourse
+): element is FirestoreSemesterPlaceholder =>
+  !!(element as FirestoreSemesterPlaceholder).startingSemester;
```

Comment:
Can you explain this?

Reply:
I got this idea from Stackoverflow, but basically it takes in an element of either type of object that can be in a semester courses list (placeholder or course), interprets it as a placeholder, and checks to see if a field unique to placeholders exists on it (like `startingSemester`). If it does, it returns true, otherwise false. 

More documentation here: https://www.typescriptlang.org/docs/handbook/advanced-types.html